pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java

```diff
@@ -377,25 +382,49 @@ public CompletableFuture<Optional<BrokerLookupData>> assign(Optional<ServiceUnit
             if (topic.isPresent() && isInternalTopic(topic.get().toString())) {
                 owner = serviceUnitStateChannel.getChannelOwnerAsync();
             } else {
-                owner = getOwnerAsync(serviceUnit, bundle, false).thenApply(Optional::ofNullable);
+                // Check if this is Heartbeat or SLAMonitor namespace
+                String candidateBroker = NamespaceService.checkHeartbeatNamespace(serviceUnit);
+                if (candidateBroker == null) {
+                    candidateBroker = NamespaceService.checkHeartbeatNamespaceV2(serviceUnit);
+                }
+                if (candidateBroker == null) {
+                    String broker = NamespaceService.getSLAMonitorBrokerName(serviceUnit);
+                    // checking if the broker is up and running
+                    if (broker != null && pulsar.getNamespaceService().isBrokerActive(broker)) {
+                        candidateBroker = broker;
```

Comment:
What if the serviceUnit is either heartbeatNamespace or SLAMonitor, but the broker is inactive? In this case, we still shouldn't assign the ownership of the namespace to other brokers.


`candidateBroker = NamespaceService.getSLAMonitorBrokerName(serviceUnit);`

Reply:
Yes, I see the original logic is to let the `SLAMonitor` assign to the active broker, and schedule unload until it assigns to the own broker.